src/pdm/cli/commands/init.py

```diff
@@ -116,7 +116,18 @@ def ask_project(self, project: Project) -> str:
     def get_metadata_from_input(self, project: Project, options: argparse.Namespace) -> dict[str, Any]:
         from pdm.formats.base import array_of_inline_tables, make_array, make_inline_table
 
-        name = self.ask_project(project)
+        if options.name:
+            if not validate_project_name(options.name):
+                project.core.ui.echo(
+                    "Project name is not valid, it should follow PEP 426",
+                    err=True,
+                    style="warning",
+                )
+                name = self.ask_project(project)
```

Comment:
I think it is expected to fail here.
You can just raise a ProjectError

Reply:
addressed